### DISCUSSION TITLE
Why UVSets are renamed ?

### DISCUSSION THREAD

**Discussion by guidoq** [Autodesk/maya-usd#2772] [Q&A]
I'm not sure why UVSets are always named st, st1, st2 etc ? If those are named with a specific name in Maya, that name should be respected. Now the original name is put in metadata.
This breaks export of USD when RenderMan for Maya is used since the RenderMan plugin honors the UVSet names (except map1 that is renamed st).
Was it a specific reasons to prompt for the renaming ?
Thanks,
Guido

**Comment by neilh-adsk** [comment]
Hi Guido,

The original submission for this change was to align closer to other environments including those at Pixar was introduced here #1528.   A later PR #2283 was introduced to allow for more customization but it didn't apply those code changes far enough.  #2776 that was submitted yesterday fixed up the remaining issues.  So you should be able to use `-preserveUVSetNames (-puv)` in order to achieve something that you are looking for.  Let me know if this solves your problems, happy to discuss ay other short comings.

Cheers,

Neil
